src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -92,27 +97,29 @@ private DedupManifestArtifactClientFactory()
             traceOutput($"Max dedup parallelism: {maxParallelism}");
 
             IDedupStoreHttpClient dedupStoreHttpClient = await AsyncHttpRetryHelper.InvokeAsync(
-                () =>
+                async () =>
                 {
                     ArtifactHttpClientFactory factory = new ArtifactHttpClientFactory(
                         connection.Credentials,
                         connection.Settings.SendTimeout,
                         tracer,
                         cancellationToken);
 
-                    IDedupStoreHttpClient client;
+                    IDedupStoreHttpClient dedupHttpclient;
                     // this is actually a hidden network call to the location service:
                     if (domainId == WellKnownDomainIds.DefaultDomainId)
                     {
-                        client = factory.CreateVssHttpClient<IDedupStoreHttpClient, DedupStoreHttpClient>(connection.GetClient<DedupStoreHttpClient>().BaseAddress);
+                        dedupHttpclient = factory.CreateVssHttpClient<IDedupStoreHttpClient, DedupStoreHttpClient>(connection.GetClient<DedupStoreHttpClient>().BaseAddress);
                     }
                     else
                     {
                         IDomainDedupStoreHttpClient domainClient = factory.CreateVssHttpClient<IDomainDedupStoreHttpClient, DomainDedupStoreHttpClient>(connection.GetClient<DomainDedupStoreHttpClient>().BaseAddress);
-                        client = new DomainHttpClientWrapper(domainId, domainClient);
+                        dedupHttpclient = new DomainHttpClientWrapper(domainId, domainClient);
                     }
 
-                    return Task.FromResult(client);
+                    this.HashType ??= await GetClientHashTypeAsync(factory, connection, client, tracer, cancellationToken);
+
+                    return await Task.FromResult(dedupHttpclient);
```

Comment:
We should be able to just return the client here right?

Reply:
Yep, missed that.